src/main/java/com/slack/eithernet/ApiResult.kt

```diff
@@ -112,6 +112,9 @@ public sealed class ApiResult<out T, out E> {
     private val HTTP_SUCCESS_RANGE = OK..299
     private val HTTP_FAILURE_RANGE = 400..599
 
+    /** Returns a new [Success] with given [value]. */
+    public fun <T : Any, E> success(value: T): ApiResult<T, E> = Success(value)
```

Comment:
If we're going to do this, let's deprecate the current constructor and add a replaceWith pointing to this. Then would need to update all the current usages to use this function too

Reply:
Should they be exclusive? I don't see any problems with being able to create a Success via this function or via the constructor (I may expect it to be the same way as the Failure case)

Reply:
Yes, if we're going to add a factory it should be the only entry point. No need for two redundant ones.

Reply:
I think the `replaceWith` cannot be done in this case (first time using it, maybe I'm missing something)

![Screenshot 2020-11-01 at 23 14 47](https://user-images.githubusercontent.com/3216712/97816802-17830000-1c98-11eb-8814-df1b0b56e633.png)

If that's the case I'll leave it up to you what to do!

Reply:
You need to apply it to the constructor, not the class